File: study/chapt2/chapt2_ws/src/demo_cpp_pkg/src/personNode.cpp
#include "rclcpp/rclcpp.hpp"
using namespace std;

class personNode : public rclcpp::Node {
private:
    string name_;
    int age_;

public:
    personNode(const string &node_name,
        const string &name,
        const int &age) : Node(node_name) {    
        this->name_ = name;
        this->age_ = age;
    }
    
    void cook(const string &food_name) {
        RCLCPP_INFO(this->get_logger(), "I am %s, and I am %d years old. "
        "I am cooking %s.", name_.c_str(), age_, food_name.c_str());
    }
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = make_shared<personNode>("cpp_node", "Larry", 23);
    node->cook("ROS2");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: study/chapt2/chapt2_ws/src/demo_python_pkg/demo_python_pkg/person_node.py
import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self, node_name: str, name: str, age: int) -> None:
        super().__init__(node_name)
        self.name = name
        self.age = age

    def eat(self, food_name: str):
        self.get_logger().info(f'我叫 {self.name}，今年 {self.age} 岁，我正在吃 {food_name}')

def main():
    rclpy.init()
    node = PersonNode('person_node', 'larry', 23)
    node.eat('fishros')
    rclpy.spin(node)
    rclpy.shutdown()